### PR TITLE
Feat/qr scanner

### PR DESCRIPTION
**Pull request by yehjxraymond**
Tradetrust Document

image

CAAS UAPL Document

image

### CODE REVIEW

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=17

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-17

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=17

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-17

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=17

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-17

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=17

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-17

File: src/services/QrProcessor.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// import Joi from "@hapi/joi";
```
Comment (sebastianquek) [inline src/services/QrProcessor.tsx]:
```suggestion
```

File: src/services/actionValidator/documentActionValidator.test.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { validateDocumentAction } from "./documentActionValidator";
+
+describe("validateDocumentAction", () => {
+  it("should not throw for valid uri", () => {
+    expect.assertions(1);
+    const input = {
+      uri: "https://test.com/doc/123"
+    };
+    expect(() => validateDocumentAction(input)).not.toThrow();
+  });
```
Comment (sebastianquek) [inline src/services/actionValidator/documentActionValidator.test.tsx]: Is it useful to add an assertion for the return value of `validateDocumentAction`?

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=17

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-17

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=17

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-17

**Pull request merged**